### PR TITLE
chore bump kakarot

### PR DESCRIPTION
**Pull request by Eikix**
Time spent on this PR:

Resolves: #<Issue number>

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/bin/hive_chain.rs [hunk 1]
```diff
@@ -60,6 +60,8 @@ async fn main() -> eyre::Result<()> {
             tokio::time::sleep(tokio::time::Duration::from_millis(500)).await;
         }
 
+        // Set the block gas limit in kakarot
```
Comment (Eikix) [inline src/bin/hive_chain.rs:63]: Should this be done in this PR or later?

Comment (greged93) [reply src/bin/hive_chain.rs:63]: I would say it can be done later if it doesn't break any hive test

Comment (greged93) [reply src/bin/hive_chain.rs:63]: Is this a TODO then?

File: src/test_utils/eoa.rs [hunk 2]
```diff
@@ -182,7 +182,7 @@ impl<P: Provider + Send + Sync> KakarotEOA<P> {
             nonce,
             max_priority_fee_per_gas: Default::default(),
             max_fee_per_gas: Default::default(),
-            gas_limit: u64::MAX,
+            gas_limit: 50_000u64,
```
Comment (greged93) [inline src/test_utils/eoa.rs:185]: Why this value?

File: src/test_utils/katana/genesis.rs [hunk 3]
```diff
@@ -203,6 +203,8 @@ impl KatanaGenesisBuilder<Loaded> {
         let proxy_class_hash = self.proxy_class_hash()?;
         let precompiles_class_hash = self.precompiles_class_hash()?;
 
+        let block_gas_limit = FieldElement::from(20_000_000u64);
```
Comment (greged93) [inline src/test_utils/katana/genesis.rs:206]: Why this value?

**Review by greged93 — COMMENTED** [review]
There are a couple of hardcoded values for the block gas limit, can we maybe make a constant for all these?

File: src/test_utils/katana/genesis.rs (hunk 3 above)
Comment (Eikix) [reply src/test_utils/katana/genesis.rs:206]: same value as kakarot

File: src/test_utils/eoa.rs (hunk 2 above)
Comment (Eikix) [reply src/test_utils/eoa.rs:185]: magic value < 20 M

File: src/eth_provider/provider.rs [hunk 4]
```diff
@@ -600,7 +600,7 @@ where
         let data = request.input.into_input().unwrap_or_default();
         let calldata: Vec<FieldElement> = data.into_iter().map_into().collect();
 
-        let gas_limit = into_via_try_wrapper!(request.gas.unwrap_or_else(|| U256::from(u64::MAX)));
+        let gas_limit = into_via_try_wrapper!(request.gas.unwrap_or_else(|| U256::from(5_000_000u64)));
```
[the remark this answers is not in the document]
Comment (Eikix) [reply src/eth_provider/provider.rs:603]: magic value < 20 M

File: src/test_utils/katana/genesis.rs (hunk 3 above)
Comment (greged93) [reply src/test_utils/katana/genesis.rs:206]: when you say Kakarot you mean what we deployed?

Comment (greged93) [reply src/test_utils/katana/genesis.rs:206]: I'm just not sure where this value comes from

File: src/eth_provider/provider.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (Eikix) [reply src/eth_provider/provider.rs:603]: changing it

**Review by greged93 — APPROVED** [review]

**Pull request merged**